File: API/Domain/Word.cs
namespace Domain
{
    public class Word
    {
        public Guid Id { get; set; }
        public string? Text { get; set; }
        public string? Definition { get; set; }
        public Guid? ParentId { get; set; }
        public Word? Parent { get; set; }
        public ICollection<Category> Categories { get; set; } = new List<Category>();
        public ICollection<Example> Examples { get; set; } = new List<Example>();
        public ICollection<Word> Synonyms{ get; set; } = new List<Word>();
    }
}

File: client/models/word.ts
export type Example = {
    id: string
    text: string
}

export type Synonym = {
    id: string
    text?: string
    isEditing: boolean
}

export type Word = {
    id: string
    text?: string
    definition?: string
    parentId?: string | null
    categories: string[]
    categoryIds?: string[]
    examples: Example[]
    synonyms: Synonym[]
    expanded: boolean
    isEditing: boolean
}

export type Category = {
    id: string;
    name: string;
}

export type WordForm = {
    id?: string;
    text: string;
    definition: string;
    parentId?: string | null;
    categoryIds: string[];
    examples: string[];
}
